base_layer/common_types/src/chain_metadata.rs

```diff
@@ -142,6 +142,47 @@ impl Display for ChainMetadata {
     }
 }
 
+/// This struct contains global reorg info
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub struct ReorgInfo {
+    /// The tip hash of the last reorg chain
+    pub last_reorg_best_block: Option<HashOutput>,
+    /// The number of blocks contained in the last reorg
+    pub num_blocks_reorged: u64,
+    /// The tip height the chain was at after the last reorg
+    pub tip_height: u64,
+}
+
+impl ReorgInfo {
+    pub fn new() -> Self {
```

Comment:
I'd prefer to not have a Default for this struct. The struct should only exist if there was a reorg. Callers can use an `Option<ReorgInfo>` and use `None` for no reorg info. The `tip_height` of `0` is a valid value and cannot be used for detecting `None`

Reply:
The `last_reorg_best_block` can then be changed to `HashOutput` and used without checking every time